app/src/main/java/nerd/tuxmobil/fahrplan/congress/settings/widgets/PreferenceListDialog.kt

```diff
@@ -0,0 +1,78 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.foundation.selection.selectableGroup
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.DialogProperties
+import kotlinx.collections.immutable.ImmutableMap
+import nerd.tuxmobil.fahrplan.congress.designsystem.buttons.ButtonText
+import nerd.tuxmobil.fahrplan.congress.designsystem.buttons.RadioButton
+import nerd.tuxmobil.fahrplan.congress.designsystem.dialogs.AlertDialog
+import nerd.tuxmobil.fahrplan.congress.designsystem.modifiers.minimumInteractiveComponentSize
+import nerd.tuxmobil.fahrplan.congress.designsystem.texts.Text
+import nerd.tuxmobil.fahrplan.congress.designsystem.themes.EventFahrplanTheme
+import kotlin.collections.forEach
+
+@Composable
+internal fun <T> PreferenceListDialog(
+    title: String,
+    entries: ImmutableMap<T, String>,
+    selectedOption: T,
+    onOptionSelected: (T) -> Unit,
+    onDismiss: () -> Unit,
+) {
+    AlertDialog(
+        title = { Text(title) },
+        text = {
+            Column(
+                modifier = Modifier
+                    .selectableGroup()
+                    .verticalScroll(rememberScrollState())
+            ) {
+                entries.forEach { (value, description) ->
+                    val isSelected = value == selectedOption
+                    Row(
+                        verticalAlignment = Alignment.CenterVertically,
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .selectable(
+                                selected = isSelected,
+                                onClick = { onOptionSelected(value) },
+                                role = Role.RadioButton
+                            )
+                            .minimumInteractiveComponentSize()
+                            .padding(horizontal = 16.dp),
+                    ) {
+                        RadioButton(
+                            selected = isSelected,
+                            onClick = null
+                        )
+                        Text(
+                            text = description,
+                            style = EventFahrplanTheme.typography.bodyLarge,
+                            modifier = Modifier.padding(start = 16.dp),
+                        )
+                    }
+                }
+            }
+        },
+        confirmButton = {
+            ButtonText(onClick = onDismiss) {
+                Text(stringResource(android.R.string.cancel))
```

Comment:
I wonder if we should rely on the Android framework to provide translations or use a custom string.

Reply:
`ConfirmationDialog` and `AlarmTimePickerFragment` are also using `android.R.string.cancel`. If you want to change this, I suggest doing it in a separate PR. 
However, I think that will only be useful if the app supports a language that Android itself doesn't support. Otherwise you'd be duplicating what's already there in Android at best. At worst, you're using different strings than users are already used to.